lib/api/truncate.js

```diff
@@ -0,0 +1,49 @@
+var SMB2Forge = require('../tools/smb2-forge');
+var SMB2Request = SMB2Forge.request;
+var BigInt = require('../tools/bigint');
+var FILE_WRITE_DATA = require('../structures/constants').FILE_WRITE_DATA;
+
+module.exports = function truncate(path, length, cb) {
+  var connection = this;
+  var file;
+  var fileLength = new BigInt(8, length);
+
+  function setFileSize(fileSizeSetted) {
+    SMB2Request(
+      'set_info',
+      {
+        FileId: file.FileId,
+        FileInfoClass: 'FileEndOfFileInformation',
+        Buffer: fileLength.toBuffer(),
+      },
+      connection,
+      function(err) {
+        if (err) cb && cb(err);
+        else fileSizeSetted();
+      }
+    );
+  }
+
+  function closeFile(fileClosed) {
+    SMB2Request('close', file, connection, function(err) {
+      if (err) cb && cb(err);
+      else {
+        file = null;
```

Comment:
Why?

Reply:
Just to clear the file from memory I guess. I've followed what was written here https://github.com/Node-SMB/marsaud-smb2/blob/master/lib/api/writeFile.js#L62

Reply:
It should not be necessary, JS garbage collector should free this variable as soon as it's no longer accessible.